### DISCUSSION TITLE
How do i generate an API key?

### DISCUSSION THREAD

**Discussion by Mrs-Feathers** [MDeLuise/plant-it#305] [Q&A]
There doesnt seem to be any information reguarding generation of an API key. i've tried using the JWT token with swagger to authenticate but that doesnt seem to work.. ive tried using the JWT token as an API key and that doesnt seem to work either. ive tried using swagger to login but i just get invalid credentials. there is no existing documentation about API other than "to get you key, follow the relevent step in relevent intructions" which i cant seem to find anywhere. i want to generate an API for homepage which it claims to support and also home assistant if i can write a quick script to get that set up.

**Comment by MDeLuise** [comment]
Hi @Mrs-Feathers, thanks for reporting this!
You're right, I apologize for the confusion caused by the lack of detailed documentation on API key generation. I will update the documentation shortly to include more specific steps. 

In the meantime, here's how you can generate an API key:
1. **Authenticate via the REST API**: Use the `/authentication/login` endpoint to log in and obtain a JWT token.
   
2. **Create the API key**: Once you have the JWT token, you can create the API key by sending a POST request to the `/api-key` endpoint. You can optionally pass a parameter like `{"name": "api-key-name"}` to name the key.

This should help you generate an API key for your homepage or any other integrations like Home Assistant.
